### PR TITLE
[Workflow] Print WorkflowInstance detail/ TaskInstance detail when finished

### PR DESCRIPTION
**Pull request by Mukvin**
## Purpose of the pull request
This pull request will print the detail of workflow and task when finished
related issue: https://github.com/apache/dolphinscheduler/issues/14346

## Brief change log

Print WorkflowInstance detail/ TaskInstance detail when finished

## Verify this pull request

UT test: org.apache.dolphinscheduler.server.master.utils.WorkflowInstanceUtilsTest

The newest log will capture as following:

```
********************************************************************************
                    Workflow Instance Detail
********************************************************************************
Workflow Name:             test_workflow
Workflow Instance Name:    test_workflow-3-20230731222903874
Command Type:              REPEAT_RUNNING
State:                     success
Host:                      192.168.140.4:5678
Is Sub Process:            no
Run Times:                 8
Max Try Times:             0
Dry Run:                   0
Tenant:                    default
Restart Time:              Tue Aug 01 10:59:29 CST 2023
Work Group:                default
Start Time:                Tue Aug 01 10:59:29 CST 2023
End Time:                  Tue Aug 01 10:59:31 CST 2023
```

```
********************************************************************************
                    Task Instance Detail
********************************************************************************
Task Name:              test_node
Workflow Instance Name: test_workflow-3-20230731222903874
Task Execute Type:      batch
Execute State:          success
Host:                   192.168.140.4:1234
Task Type:              SHELL
Priority:               medium
Tenant:                 default
First Submit Time:      Tue Aug 01 10:59:29 CST 2023
Submit Time:            Tue Aug 01 10:59:29 CST 2023
Start Time:             Tue Aug 01 10:59:29 CST 2023
End Time:               Tue Aug 01 10:59:30 CST 2023

```

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java [hunk 1]
```diff
@@ -805,6 +809,65 @@ public void checkSerialProcess(ProcessDefinition processDefinition) {
         commandService.createCommand(command);
     }
 
+    private void logTaskInstanceInDetail(TaskInstance taskInstance) {
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java]: Please move this kind of method to utils

Comment (Mukvin) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java]: done.

**Review by ruanwenjun — DISMISSED** [review]

**Review by fuchanghai — APPROVED** [review]

**Review by rickchengx — APPROVED** [review]
LGTM, rerun failed CI

**Review by davidzollo — APPROVED** [review]
+1

**Pull request merged**